spec/services/partner_routing_service_spec.rb

```diff
@@ -61,12 +61,72 @@
 
       context "when clients zip code doesn't correspond to a Vita Partner" do
         context "when state for that zip code has associated Vita Partners" do
-          subject { PartnerRoutingService.new(zip_code: "28806") }
+          subject { PartnerRoutingService.new(zip_code: "28806") } #NC
 
           let!(:expected_vita_partner) { create :vita_partner }
 
+          context "with state-qualified organizations with excess capacity" do
+            let!(:org_with_capacity_and_routing_1) {
+              vita_partner = create(:vita_partner, capacity_limit: 10)
+              create(:vita_partner_state, state: "NC", routing_fraction: 0.3, vita_partner: vita_partner)
+              vita_partner
+            }
+            let!(:org_with_capacity_and_routing_2) {
+              vita_partner = create(:vita_partner, capacity_limit: 10)
+              create(:vita_partner_state, state: "NC", routing_fraction: 0.3, vita_partner: vita_partner)
+              vita_partner
+            }
+            let!(:org_with_routing_without_capacity) {
+              vita_partner = create(:vita_partner, capacity_limit: 0)
+              create(:vita_partner_state, state: "NC", routing_fraction: 0.3, vita_partner: vita_partner)
+              vita_partner
+            }
+            let!(:org_with_routing_without_excess_capacity) {
+              vita_partner = create(:vita_partner, capacity_limit: 5)
+              create(:vita_partner_state, state: "NC", routing_fraction: 0.3, vita_partner: vita_partner)
+              (vita_partner.capacity_limit + 1).times do
+                create :client_with_status, vita_partner: vita_partner, status: "intake_ready"
+              end
+            }
+            before do
+              weighted_service_double = double(WeightedRoutingService)
+              allow(WeightedRoutingService).to receive(:new).and_return weighted_service_double
+              allow(weighted_service_double).to receive(:weighted_routing_ranges).and_return(
+                [
+                  { id: 1, low: 0.0, high: 0.2 },
```

Comment:
We tested with an empty array returned for weighted_routing_ranges and it still worked--consider using the empty array?